tonilopezmr/day4/PassphraseValidator.kt

```diff
@@ -0,0 +1,73 @@
+package day4
+
+import org.junit.Assert.*
+import org.junit.Test
+import java.io.File
+import java.util.*
+
+class PassphraseValidator {
+
+  fun validI(passphrase: String): Boolean {
+    val words = passphrase
+        .split(" ")
+    return words
+        .foldIndexed(true) { idx, acc, a ->
+          if (words.subList(idx + 1, words.size).contains(a)) {
+            false
+          } else {
+            acc
+          }
+        }
+  }
+
+  fun valid(passphrase: String): Boolean {
+    val list = LinkedList(passphrase.split(" "))
+    val element = list.poll()
```

Comment:
I suppose you are using `LinkedList` to get `Dequeue` behaviour. But did you know you could the same with an immutable list? (Just a bit uglier though, but in Kategory they have implemented methods for `(x : xs)` haskell notation like)

```kotlin
val list = listOf(1,2,3)
val (head, tail) = Pair(list.first, list.drop(1))
// head -> 1
// tail -> listOf(2,3)
```

Reply:
You are droping in the same way, Yes I did with LinkedList to work like (x :xs)

Reply:
Except, `drop` is immutable. It returns a new list.

Reply:
Yes, In kategory, we have several issues with OutOfMemory with drop in `ListKW.fold` for many elements 👎